internal/codeintel/autoindexing/internal/store/store_indexes.go

```diff
@@ -472,16 +550,67 @@ func (s *store) IsQueued(ctx context.Context, repositoryID int, commit string) (
 	}})
 	defer endObservation(1, observation.Args{})
 
-	count, _, err := basestore.ScanFirstInt(s.db.Query(ctx, sqlf.Sprintf(isQueuedQuery, repositoryID, commit, repositoryID, commit)))
-	return count > 0, err
+	isQueued, _, err := basestore.ScanFirstBool(s.db.Query(ctx, sqlf.Sprintf(
+		isQueuedQuery,
+		repositoryID, commit,
+		repositoryID, commit,
+	)))
+	return isQueued, err
 }
 
 const isQueuedQuery = `
-SELECT COUNT(*) WHERE EXISTS (
-	SELECT id FROM lsif_uploads_with_repository_name WHERE repository_id = %s AND commit = %s AND state NOT IN ('deleted', 'deleting')
-	UNION
-	SELECT id FROM lsif_indexes_with_repository_name WHERE repository_id = %s AND commit = %s
-)
+SELECT
+	EXISTS (
+		SELECT 1
+		FROM lsif_uploads u
+		WHERE
+			repository_id = %s AND
+			commit = %s AND
+			state NOT IN ('deleting', 'deleted') AND
+			associated_index_id IS NULL
+	)
+
+	OR
+
+	EXISTS (
+		SELECT 1
+		FROM (
+			SELECT DISTINCT ON (root, indexer) should_reindex
+			FROM lsif_indexes
+			WHERE repository_id = %s AND commit = %s
+			ORDER BY root, indexer, queued_at DESC
+		) _
+		WHERE NOT should_reindex
+	)
```

Comment:
This got me suspicious, please hold.

Reply:
```suggestion
	-- We want IsQueued to return true when there exists auto-indexing job records
	-- and none of them are marked for reindexing. If we have one or more rows and
	-- ALL of them are not marked for re-indexing, we'll block additional indexing
	-- attempts.
	SELECT COALESCE(bool_and(NOT should_reindex), false)
	FROM (
		-- For each distinct (root, indexer) pair, use the most recently queued 
		-- index as the authoritative attempt.
		SELECT DISTINCT ON (root, indexer) should_reindex
		FROM lsif_indexes
		WHERE repository_id = %s AND commit = %s
		ORDER BY root, indexer, queued_at DESC
	) _
```		

I think the original code would require ALL of the `should_reindex` field to be set, where we want only one to trigger a re-indexing (the `IsQueuedRootIndexer` is meant to catch the duplicates when there's a mix of should-reindex and shouldn't-reindex records in the same result set).